### ISSUE TITLE
Compare performance for transformed emulator and current pipeline

### ISSUE THREAD

**Issue by sgreenbury** [alan-turing-institute/autoemulate#618]
In particular, the reaction diffusion example appears to have differences in performance of the final fitted emulator.

**Comment by sgreenbury** [comment]
Thanks @ContiPaolo for adding the results for comparison here!

One other potential difference I was wondering about could be if the `TransformedEmulator` is being differently to the `AutoEmulatePipeline`. The `TransformedEmulator` is fit directly on transformed versions of `x` and `y` (i.e. `model.fit(x_transformed, y_transformed)`). This means loss calculations and parameter updates are in the transformed space. I think this is the case for the `AutoEmulatePipeline` too?

**Comment by sgreenbury** [comment]
> One other potential difference I was wondering about could be if the `TransformedEmulator` is being fit differently to the `AutoEmulatePipeline`. The `TransformedEmulator` is fit directly on transformed versions of `x` and `y` (i.e. `model.fit(x_transformed, y_transformed)`). This means loss calculations and parameter updates are in the transformed space. I think this is the case for the `AutoEmulatePipeline` too?

Having looked further into the docs the `TransformedEmulator` does implement fitting in the same way as the sklearn version ([docs](https://scikit-learn.org/stable/modules/generated/sklearn.compose.TransformedTargetRegressor.html?utm_source=chatgpt.com#transformedtargetregressor)). So it is likely to be hyperparameter search for the GP.

**Comment by ContiPaolo** [comment]
After some exploration, I might have found the main issue: the v0 version was standardizing the target, while v1 was not, which seems to be a significant problem for the GP.

I created a new branch `exploreGP` where I introduced this standardization in the [GaussianProcessExact](https://github.com/alan-turing-institute/autoemulate/blob/exploreGP/autoemulate/experimental/emulators/gaussian_process/exact.py). Now, the tutorials for reaction diffusion v0 and v1 produce exactly the same results in terms of the mean, without requiring any tuning.

Hopefully, this mostly solves the issue. However, it is still unclear why `StandardizeTransform()` actually performs standardization on the targets $y$. We also need to double-check how this standardization affects the covariance, since different versions show different performance in terms of standard deviation.ces in terms of std).
